### ISSUE TITLE
[FEATURE] Allow custom tags for underlying cluster on databricks_sql_permissions resource or allow a custom cluster_id

### ISSUE THREAD

**Issue by psg2** [databrickslabs/terraform-provider-databricks#678]
Hi there,

Similar to a problem that I mentioned on another [issue](https://github.com/databrickslabs/terraform-provider-databricks/issues/654), at my company we have AWS policies enforcing some tag rules across all AWS accounts for EC2 instances, and for that I would like to specify some tags for the underlying cluster managed by the SQL permissions resource and this should help not only to bypass the policies but to get better cost tracking for this cluster, ideally it should be minimal but it would be good to track it.

So in short, I would like to either have a custom_tags attribute on the databricks_sql_permissions resource or the possibility to specify a cluster_Id like on the databricks_aws_s3_mount resource.

Code: https://github.com/databrickslabs/terraform-provider-databricks/blob/71bd7f63080a5f447b7af1780ca41b89996bf20d/access/resource_sql_permissions.go#L260

What do you think about this, Is it a good addition for the provider? If so I can make the pull request for this too.

**Comment by nfx** [comment]
Cluster_id is there, just not documented yet

**Comment by psg2** [comment]
You're absolutely right, I read the code but for a moment I misunderstood thinking that because it had the computed annotation we wouldn't be able to set it up and didn't even try it out prior to opening the issue. 😓 

My bad here.

**Comment by nfx** [comment]
The simplest thing you can do is start using cluster_id and PR few lines to documentation :)

I really wonder how it slipped my mind in today's release.

Looking forward to feedback on this particular resource at your scale.

**Comment by psg2** [comment]
TBH, we barely use Table Access Control at Wildlife yet, because, without an abstraction layer to manage it like this terraform resource, I always thought it wasn't a good experience to manage it manually from spark clusters.

So first of all, thank you for providing this resource, we're barely starting to use table ACL and this will likely drive more adotption.

**Comment by nfx** [comment]
@psg2 if field is computed, it doesn't mean it cannot be set :)

This resource is one of the most complex ones. Waiting for REST API to be released for Unity Catalog, so that we can replace command/context flow here.

**Comment by nfx** [comment]
Also keep in mind that updates to table acls are not atomic - e.g. first all permissions are revoked and then only those configured are applied. 

It would be very helpful if you blog/tweet about this SQL permissions resource. I've mentioned it on DAIS talk, did spread enough word internally, but it doesn't get enough adoption as I think it should.

**Comment by psg2** [comment]
> @psg2 if field is computed, it doesn't mean it cannot be set :)

I guess this is the first time I actually use a resource with Computed as optional.

> This resource is one of the most complex ones. Waiting for REST API to be released for Unity Catalog, so that we can replace command/context flow here.

Really looking forward to that as well, loved the announcement during the summit.

> Also keep in mind that updates to table acls are not atomic - e.g. first all permissions are revoked and then only those configured are applied.

Oh, that's very nice to know.

> It would be very helpful if you blog/tweet about this SQL permissions resource. I've mentioned it on DAIS talk, did spread enough word internally, but it doesn't get enough adoption as I think it should.

I will keep you posted on this as well, I'm personally not the best when it comes to social media. 😅 

Also, I watched your presentation on DAIS, good job there. Awesome to see how other companies are using the provider, after the talk I spent a good amount of time reading the scribd tech blog.

**Comment by nfx** [comment]
@psg2 can this issue be closed then? :)

**Comment by psg2** [comment]
> @psg2 can this issue be closed then? :)

I've managed to use the resource yesterday providing a cluster id, but before that it failed because it was trying to create a high concurrency cluster without workers which is basically the underlying cluster that the provider tries to create. 🤔

But I guess we can close this one and I will try to reproduce this error again tomorrow and might open a proper issue about it instead of reporting here.

**Comment by nfx** [comment]
@psg2 High concurrency means "with additional security". It technically could be done on a single node as well. I think it's failing in your environment because of no tags specified. Exactly the same semantics as in mount resources.
